File: Dart/392.is_subsequence.dart
void main() {
  print(isSubsequence("abc", "ahbgdc"));
  print(isSubsequence("axc", "ahbgdc"));
  print(isSubsequence("", "ahbgdc"));
  print(isSubsequence("b", "abc"));
}

bool isSubsequence(String s, String t) {
  if (s.isEmpty) {
    return true;
  }

  var index = 0;

  for (int i = 0; i < t.length; i++) {
    if (index >= s.length) {
      break;
    } else if (s[index] == t[i]) {
      index++;
    }
  }

  return index == s.length;
}

File: src/392.is_subsequence.kt
fun main() {
    println(isSubsequence("abc", "ahbgdc"))
    println(isSubsequence("axc", "ahbgdc"))
    println(isSubsequence("", "ahbgdc"))
    println(isSubsequence("b", "abc"))
}

fun isSubsequence(s: String, t: String): Boolean {
    if (s.isEmpty()) {
        return true
    }

    var index = 0

    for (element in t) {
        if (index >= s.length) {
            break
        } else if (s[index] == element) {
            index++
        }
    }

    return index == s.length
}
